### DISCUSSION TITLE
zksync-ethers: A Unified Package for Ethers.js Integration

### DISCUSSION THREAD

**Discussion by idea404** [zkSync-Community-Hub/zksync-developers#233] [Announcements]
# Introducing **`zksync-ethers`**: A Unified Package for Ethers.js Integration

We are excited to announce an important update to our JavaScript SDK packages that wrap the `ethers` library. To streamline our offerings and enhance clarity, we are renaming and unifying our existing packages, **`zksync-web3`** and **`zksync2-js`**, to  **`zksync-ethers`**.

## 💻 **What's Changing?**

*Ethers 6*

- The package known as **`zksync2-js`** (wrapper for Ethers v6) will merge into **`zksync-ethers`** as version **`6`**. Install it with `npm i zksync-ethers`.
- We kindly ask new Issues and Feature Requests be created on the [`zksync-ethers` repository](https://github.com/zksync-sdk/zksync-ethers).

*Ethers 5*

- The package currently known as **`zksync-web3`** (wrapper for Ethers v5) will now be part of **`zksync-ethers`** as version **`5`**. Install it with `npm i zksync-ethers@5`.
- We will continue to support version **`5`** as long as the main Ethers 5 package is actively maintained.

## 🙋 **Why This Change?**
We aim to simplify the developer experience and provide a more intuitive package naming system. By consolidating our SDKs under a single name, we make it easier for developers to find and use the right tools for their projects on zkSync.

## 🔧 **Impact on Your Projects**

- New versions of the `hardhat-zksync` plugins will be released using the new SDK.

*Ethers 6*

- Updating the SDK if you’re using `zksync2-js`:
    
    ```bash
    npm uninstall zksync2-js
    npm install zksync-ethers
    ```
    
- Updating plugins will not require an immediate update if your project uses the latest versions of these plugins which are dependent on Ethers 6.

*Ethers 5*

- Updating the SDK if you’re using `zksync-web3`:
    
    ```bash
    npm uninstall zksync-web3
    npm install zksync-ethers@5
    ```
    
- Plugins will have a minor release (e.g., from `0.15.0` to `0.16.0`) that updates the dependency to fetch `zksync-ethers@5`. The tag `legacy-ethers-v5` will be used for plugin releases on NPM using Ethers 5.

## ⏰ **Timeline**

- **`zksync-ethers`** is now available (as of *Friday, December 8th, 2023*)
- **`zksync-web3`** and **`zksync2-js`** will throw deprecation warnings pointing to **`zksync-ethers`**
- Docs, READMEs, and NPM sites will include deprecation messages
- Both **`zksync-web3`** and **`zksync2-js`** will be supported for patch versions until *Monday, January 8th, 2023*, after which new versions will only be released for **`zksync-ethers`**

## 📩 **Support and Feedback**
We understand that changes like these can raise questions and concerns. Our team is here to assist you through this transition. Please reply to this thread if you have any feedback, questions or need support.

**Comment by Morgosh** [comment]
<img width="738" alt="image" src="https://github.com/zkSync-Community-Hub/zkync-developers/assets/35894423/724f705f-6ac2-42b1-9ee9-15bcb5632eb6">
when tying new ethers 6 + zksync-ethers + tutorial local env greeter

**Reply by Morgosh** [reply]
"@matterlabs/hardhat-zksync-deploy": "^1.1.0",
    "@matterlabs/hardhat-zksync-solc": "^1.0.3",
    "@matterlabs/hardhat-zksync-verify": "^1.2.0",
    "zksync-ethers": "^6.0.0"
    "ethers": "^6.9.0",

**Reply by zkssui9** [reply]
掏腰包

**Comment by Morgosh** [comment]
and then for compiling this error:

Class extends value undefined is not a constructor or null

**Comment by dzhamilrustamov** [comment]
Good

**Comment by HoseinNikmaramOfficial** [comment]
thanks mate 💯

**Comment by Ea1154** [comment]
nice mate

**Comment by eseang** [comment]
good project

**Comment by eseang** [comment]
I really love your project When without or deposited it so fast and fee so cheap However, I've encountered occasional delays of up to 24 hours when facing network issues. I urge the ZKSync team to address these concerns before the mainnet launch to ensure a smoother user experience. I remain confident in the team's abilities and the project's potential to transform blockchain technology.

**Comment by jablonskibrittany** [comment]
This update streamlines the zkSync JavaScript SDK packages by merging zksync-web3 and zksync2-js into a unified package called zksync-ethers. Developers can now install zksync-ethers for both Ethers v5 and v6 integration, simplifying package management and providing a clearer naming convention. The transition

**Comment by berronesjosh** [comment]
Announcement: Exciting update! zksync-web3 and zksync2-js are merging into zksync-ethers, simplifying the developer experience with a unified package for Ethers integration. Upgrade now for streamlined development on zkSync!

**Comment by dgimenes12** [comment]
very good team is working hard to be a better blockchain solution

**Comment by sarahzitzner** [comment]
Unified zksync-ethers package introduced for seamless integration with Ethers.js. Simplify development and enhance clarity on zkSync.

**Comment by dgimenes12** [comment]
Yes nice congrats nice projects

**Comment by develei** [comment]
nice! This looks like a very useful project.

**Comment by deanhobson** [comment]
The JavaScript SDK packages for zkSync are being renamed and unified under the new name zksync-ethers. This change aims to simplify the developer experience and provide a more intuitive package naming system, making it easier for developers to find and use the right tools for their projects on zkSync. The zksync2-js package will merge into zksync-ethers as version 6, while the zksync-web3 package will now be part of zksync-ethers as version 5.

**Comment by ldilago** [comment]
Great job
